include/object_constants.h

```diff
@@ -198,947 +200,2231 @@
     #define MOV_BCOIN_ACT_STOPPED                           0x3
     #define MOV_BCOIN_ACT_FLICKERING                        0x4
 
-/* Bob-omb */
+/* Yellow Coin */
     /* oBehParams2ndByte */
-    #define BOBOMB_BP_STYPE_GENERIC    0x0
-    #define BOBOMB_BP_STYPE_STATIONARY 0x1
-    /* oAction */
-    #define BOBOMB_ACT_PATROL      0x0
-    #define BOBOMB_ACT_LAUNCHED    0x1
-    #define BOBOMB_ACT_CHASE_MARIO 0x2
-    #define BOBOMB_ACT_EXPLODE     0x3
+    #define YELLOW_COIN_BP_ONE_COIN                         0x1
 
 /* Moving Yellow Coin */
     /* oAction */
-    #define MOV_YCOIN_ACT_IDLE 0
-    #define MOV_YCOIN_ACT_BLINKING 1
+    #define MOV_YCOIN_ACT_IDLE                              0x0
+    #define MOV_YCOIN_ACT_BLINKING                          0x1
+    /* oSubAction */
+
+/* Coin Formation */
+    /* (oBehParams2ndByte & 0x7) */
+    #define COIN_FORMATION_BP_HORIZONTAL_LINE               0x0
+    #define COIN_FORMATION_BP_VERTICAL_LINE                 0x1
+    #define COIN_FORMATION_BP_HORIZONTAL_RING               0x2
+    #define COIN_FORMATION_BP_VERTICAL_RING                 0x3
+    #define COIN_FORMATION_BP_ARROW                         0x4
+    /* oAction */
+    #define COIN_FORMATION_ACT_INACTIVE                     0x0
+    #define COIN_FORMATION_ACT_ACTIVE                       0x1
+    #define COIN_FORMATION_ACT_DEACTIVATE                   0x2
 
-/* Bob-omb Buddy */
+/* Coin Inside Boo */
```

Comment:
Every change on this file could really hurt compatibility. We'll need a full revert of this file, but keep `OBJ_FLAG_OCCLUDE_SILHOUETTE` and `OBJ_FLAG_OPACITY_FROM_CAMERA_DIST`, and any other 100% necessary changes for features

Reply:
blocking

Reply:
damn i thought this was reordering/formatting and not documentation. this is seriously great stuff for documentation and should probably be requested for base decomp. that said, changes are just too heavy for here imo but i'm down to talk this out.

Reply:
The currently unused defines can probably be reverted, and I can re-add them with one commit per behavior file if needed.

Reply:
the acts seem like an enum per file kind of thing imo but its fine, i'll resolve this comment but _please no more changes to this massive file I cant handle it_